unit-tests/src/test/scala/scala/PrimitiveSuite.scala

```diff
@@ -50,4 +50,22 @@ object PrimitiveSuite extends tests.Suite {
     val y: Long = 33
     assert((x << y) == 6)
   }
+
+  test(" x % y") {
+    assert(intRemByNegOne(4) == 0)
+    assert(intRemByNegOne(Int.MinValue) == 0)
+    assert(longRemByNegOne(4) == 0L)
+    assert(longRemByNegOne(Long.MinValue) == 0L)
+  }
+
+  // use method to prevent partial evaluation
+  def intRemByNegOne(i: Int): Int = {
+    i % -1
+  }
+
+  // use method to prevent partial evaluation
+  def longRemByNegOne(l: Long): Long = {
+    l % -1L
+  }
```

Comment:
Actually we want to test both the case when partial evaluation happens and the case when it doesn't. To do this you can define two duplicate definitions with the same body, one marked as `@inline` and another one as `@noinline`.

Reply:
Those small methods should probably be defined within `test("...") { ... }` as well.

Reply:
Done